cluster/src/main/java/io/scalecube/cluster/membership/MembershipProtocolImpl.java

```diff
@@ -91,27 +90,28 @@
    * @param failureDetector failure detector
    * @param gossipProtocol gossip protocol
    * @param config membership config parameters
+   * @param scheduler scheduler
    */
   public MembershipProtocolImpl(
       AtomicReference<Member> memberRef,
       Transport transport,
       FailureDetector failureDetector,
       GossipProtocol gossipProtocol,
-      MembershipConfig config) {
+      MembershipConfig config,
+      Scheduler scheduler) {
+
     this.transport = Objects.requireNonNull(transport);
     this.config = Objects.requireNonNull(config);
     this.failureDetector = Objects.requireNonNull(failureDetector);
     this.gossipProtocol = Objects.requireNonNull(gossipProtocol);
     this.memberRef = Objects.requireNonNull(memberRef);
-
-    Member member = memberRef.get();
-    String nameFormat = "sc-membership-" + Integer.toString(member.address().port());
-    this.scheduler = Schedulers.newSingle(nameFormat, true);
+    this.scheduler = Objects.requireNonNull(scheduler);
 
     // Prepare seeds
     this.seedMembers = cleanUpSeedMembers(config.getSeedMembers());
 
     // Init membership table with local member record
+    Member member = memberRef.get();
     MembershipRecord localMemberRecord = new MembershipRecord(member, ALIVE, 0);
```

Comment:
Maybe we need to change MembershipRecord like
```java
new MembershipRecord(member::get, ALIVE, 0);
```
because we can store an old member data